daffodil-test/src/test/java/org/goodudfs/example/StringFunctions/StringFunctionsProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.goodudfs.example.StringFunctions;
+
+import org.apache.daffodil.udf.UserDefinedFunctionProvider;
+
+public class StringFunctionsProvider extends UserDefinedFunctionProvider {
+
+  @Override
+  public Class<?>[] getUserDefinedFunctionClasses() {
+    // TODO Auto-generated method stub
+    return new Class<?>[] { Replace.class, Compare.class };
+  }
+}
```

Comment:
Since these are good examples and we'll probably point people to them, it might also be useful to have a scala provider and UDF as well.

Reply:
So trying to implement this has me revisting the fromClass function. Should we allow UDFs to have param/return types that are iterables/collections of types we support?

```scala
def evaluate(stringsToJoin: Array[String], sep: String) = {...}
```

Reply:
I'd say in this first pass keep things simple and don't worry about it. We can always add that capability later if needed.